src/particles/media/component.tsx

```diff
@@ -29,11 +29,39 @@ export const Media = (props: IMediaProps): React.ReactElement => {
     return fileExtension === 'mp4' || fileExtension === 'webm' || fileExtension === 'ogg';
   }, [props.source]);
 
-  return isVideo ? (
+  React.useEffect((): void => {
+    if (!props.source) {
+      setMediaType(null);
+      return;
+    }
+    fetch(props.source)
+      .then((response) => {
+        if (response.status >= 300) {
+          throw new Error(`Failed to fetch content type: ${response}`);
+        }
+        const contentType = response.headers.get('Content-Type');
+        if (!contentType) {
+          setMediaType(null);
+          return;
+        }
+        const arr = contentType.split('/'); // ['image', 'jpeg'] | ['video', 'mp4']
+        if (arr.length === 0) {
+          setMediaType(null);
+          return;
+        }
+        setMediaType(arr[0]);
+      })
+      .catch(() => {
+        setMediaType(null);
+      });
+    // setMediaType(media);
```

Comment:
removable?

Reply:
I will say no since it will throw an uncaught error if the extension and link type isn't known. I can explain a scenario

Reply:
oh i mean the commented line here